### PR TITLE
[tvOS] Fix play/pause button of siri remote

### PR DESCRIPTION
**Pull request by sy6sy2**
Restore the Siri Remote play/pause button handling in main.

### CODE REVIEW

**Review by LePips — CHANGES_REQUESTED** [review]
This actually requires a lot more state handling.

1 - Don't make a new state handler, the player state is held in the environment `VideoPlayerManager` object. The behavior should be the same as the play buttons based on the player state for iOS.
2 - We need to set the state of the overlays. When the play/pause button is pressed, the main overlay should be presented if it isn't and poke the overlay timer.

File: Swiftfin tvOS/Views/VideoPlayer/Overlays/Overlay.swift [hunk 1]
```diff
@@ -26,6 +26,8 @@ extension VideoPlayer {
         private var confirmCloseWorkItem: DispatchWorkItem?
         @State
         private var currentOverlayType: VideoPlayer.OverlayType = .main
+        @State
+        private var isPlaying = true
```
Comment (jhays) [inline Swiftfin tvOS/Views/VideoPlayer/Overlays/Overlay.swift:30]:
```suggestion
@EnvironmentObject
private var videoPlayerManager: VideoPlayerManager
```

File: Swiftfin tvOS/Views/VideoPlayer/Overlays/Overlay.swift [hunk 2]
```diff
@@ -69,6 +71,15 @@ extension VideoPlayer {
...
                     overlayTimer.start(5)
                 }
+                .onPlayPauseCommand(perform: {
+                    if isPlaying {
+                        proxy.pause()
+                        isPlaying = false
+                    } else {
+                        proxy.play()
+                        isPlaying = true
+                    }
```
Comment (jhays) [inline Swiftfin tvOS/Views/VideoPlayer/Overlays/Overlay.swift:81]:
```suggestion
                    if !isPresentingOverlay {
                        isPresentingOverlay = true
                        overlayTimer.start(5)
                    }
                    switch videoPlayerManager.state {
                    case .playing:
                        proxy.pause()
                    default:
                        proxy.play()
                    }
```

File: Swiftfin tvOS/Views/VideoPlayer/Overlays/Overlay.swift (hunk 1 above)
Comment (jhays) [reply Swiftfin tvOS/Views/VideoPlayer/Overlays/Overlay.swift:30]: Probably best to place this `@EnvironmentObject` above with the other environment objects, its just that I couldn't find a way to use GitHub's suggestion feature on lines that have not been changed already.

File: Swiftfin tvOS/Views/VideoPlayer/Overlays/Overlay.swift (hunk 2 above)
Comment (jhays) [reply Swiftfin tvOS/Views/VideoPlayer/Overlays/Overlay.swift:81]: @LePips does this solve your change request?

File: Swiftfin tvOS/Views/VideoPlayer/Overlays/Overlay.swift (hunk 1 above)
Comment (LePips) [reply Swiftfin tvOS/Views/VideoPlayer/Overlays/Overlay.swift:30]: Yes, ti should go with the other `EnvironmentObject`s.

File: Swiftfin tvOS/Views/VideoPlayer/Overlays/Overlay.swift (hunk 2 above)
Comment (LePips) [reply Swiftfin tvOS/Views/VideoPlayer/Overlays/Overlay.swift:81]: Not just seeing if the overlay is being presented but also set it to the main overlay.

**Comment by LePips** [comment]
This will be overridden by #1066 with the change to not use the SwiftUI press handlers and use `PreferencesView` instead.

See why necessary:
- https://github.com/jellyfin/Swiftfin/pull/1066#discussion_r1614035275

**Pull request closed without merging**